### ISSUE TITLE
wiki vs docs/

### ISSUE THREAD

**Issue by zbraniecki** [unicode-org/icu4x#292]
Most of our docs are checked in into `docs/` directory.

I made an attempt to try wiki instead with my data provider tutorial.

As we are nearing 0.1 release and I start looking at the funnels and how users will learn about our project I am looking to improve our readmes, docs and wikis and I'm wondering about how to properly reason about split between the role of the wiki and docs.

Since we started working on icu4x I see the main benefit of docs being a thorough review of each document performed before it gets committed.
After that, both docs and wiki keep history of changes, and the main difference seems to be that every change to docs has a higher organizational overhead as it requires a PR and review.

Is it worth it?

I like the simplicity of wiki, and I trust each one of us that as we iterate over wiki we'll make reasonable changes and we can discuss them in issues. The wiki seems to me like a lower-overhead model with better interlinking and low-cost updates.

What is the groups position? When and why do we prefer docs over wiki and do we want both, or coalesce on one? If we keep both can we find a rule of thumb on when to use which?

**Comment by EvanJP** [comment]
#### Previous Works:
I looked up to see what other thoughts on the matter have been and found [this previous github discussion from the Microsoft terminal repo ](https://github.com/microsoft/terminal/issues/636), [this one from a SAAS company](https://github.com/cdr/code-server/issues/401), and [this blog post](https://www.dzombak.com/blog/2015/03/Why-use-GitHub-Wikis-.html). [This blog post](https://www.growingwiththeweb.com/2016/07/enabling-pull-requests-on-github-wikis.html) discusses the idea of using PRs for github wikis to resolve issues around branching and such, but seems like a hassle as well.

TLDR: Wiki only represents latest release (have to manually go through wiki history), while docs allow for easier branching/snapshotting per versions of the project. Looking at the discussions above, people have decided to just stick with a `docs` folder. 

#### Personal Opinion:
I am a little worried that keeping both the docs and wiki updated with each other might be more hassle than it's worth if we decide to do both (more like one person forgets to update one but updates the other, and it gets confusing for users). 

I think unless docs need to be changed consistently we should just stick with the `docs` folder rather than the wiki.

#### Side Note: 
I've been meaning to ask this for months, but for some reason never brought it up: Is there a specific reason why `index.md` is not `README.md` to serve as a somewhat of a home page for the `docs/` directory? It seems a little odd as the `index.md` seems to serve the purpose that a `README.md` is supposed to do, yet feels out of the way as there is no obvious indication that someone should start there.

**Comment by sffc** [comment]
Things under `docs` are more like design documents than user guides.  They are official documents approved by the ICU4X-SC.  This is written in our charter.

We can put user guide material on the wiki.  Alternatively, why can't we stick with rustdoc and possibly README files for user guides?  I find wikis tend to get stale really fast.

**Comment by zbraniecki** [comment]
The benefit of rustoc would be that API examples would get tested. I think there's also a tool to cover md files with tests.

I'm writing a doc on how to structure benchmarks in criterion. It doesn't really fit the rustdoc too much and it's not any official bylaws or charter doc.
It's more of a tutorial like DataProvider one is on wiki.
Would you prefer to coalesce both in docs/ as md files?

**Comment by sffc** [comment]
Put it on the wiki for now, and let's discuss our docs strategy in an upcoming meeting.

**Comment by zbraniecki** [comment]
Added https://github.com/unicode-org/icu4x/wiki/Benchmarking - please review!

**Comment by sffc** [comment]
I'm increasingly disliking adding these pages to the wiki, especially pages that have code.  Updating the wiki requires careful testing that the code compiles.  It would be nicer if we could run tooling to automatically build and test the example code.

**Comment by filmil** [comment]
A somewhat surprising, but seemingly working way to use *existing* tools to ensure examples and inline code compile and run, is here: https://docs.rs/dtolnay/0.0.9/dtolnay/macro._01__await_a_minute.html

tl;dr: using `rustdoc` as a blog.

**Comment by zbraniecki** [comment]
I also prefer `md` files, and I know that there's a project that makes cargo test test the `md` files which would be super useful.
